### PR TITLE
Issue #3236 Update attribute requirement levels in hardware metrics semconv

### LINKED ISSUE #3236: Review existing "optional" attributes in hardware metrics
from @bertysentry's https://github.com/open-telemetry/opentelemetry-specification/pull/3228#discussion_r1111801889:

> The exact definition of Optional (i.e. really opt-in) was provided after I started working on the semantic conventions for hardware. Which lead to this confusion.

### PR DESCRIPTION
**Pull request by bertysentry**
Fixes #3236

## Changes

* Update attributes requirement levels in `specification\metrics\semantic_conventions\hardware-metrics.md`
* Minor wording corrections

### CODE REVIEW

**Comment by bertysentry** [comment]
@trask There it is! 😉

File: specification/metrics/semantic_conventions/hardware-metrics.md [hunk 1]
```diff
@@ -38,34 +38,33 @@ when creating instruments not explicitly defined in the specification.
 All metrics in `hw.` instruments should be attached to a [Host Resource](../../resource/semantic_conventions/host.md)
 and therefore inherit its attributes, like `host.id` and `host.name`.
 
-Additionally, it is recommended that all metrics in `hw.` instruments have the
-below attributes:
+Additionally, all metrics in `hw.` instruments should have the below attributes:
```
Comment (trask) [inline specification/metrics/semantic_conventions/hardware-metrics.md:41]:
```suggestion
Additionally, all metrics in `hw.` instruments have the following attributes:
```

**Review by trask — APPROVED** [review]

File: specification/metrics/semantic_conventions/hardware-metrics.md (hunk 1 above)
Comment (bertysentry) [reply specification/metrics/semantic_conventions/hardware-metrics.md]: done, thx!

**Comment by bertysentry** [comment]
Just rebased on *main* and resolved conflicts in `CHANGELOG.md`

**Comment by bertysentry** [comment]
Rebased **again** and resolved conflicts **again** 🙄

**Review by trask — APPROVED** [review]

**Comment by trask** [comment]
@open-telemetry/technical-committee is this ready to merge? thx

**Comment by bertysentry** [comment]
Yes, please merge, as I don't want to keep solving conflicts in `CHANGELOG.md`... 🙄 Thank you!

**Comment by jack-berg** [comment]
@bertysentry sorry about the delay. I'm happy to merge if you can merge main one more time.

**Comment by bertysentry** [comment]
@jack-berg Rebase done! Thank you!

**Comment by bertysentry** [comment]
@jack-berg markdown-link-check just failed, but it's not related to this PR. Could you please rerun the checks? Thanks!

**Comment by jack-berg** [comment]
@bertysentry unfortunately the build was temporarily broken when you resolved the merge conflicts. Its since been resolved with #3327 so you'll need to resolve the conflicts once more.

Normally, maintainers can resolve trivial merge conflicts right before merging the PR. I think that's not possible in this case because the fork has more restrictive branch permissions.

File: specification/metrics/semantic_conventions/hardware-metrics.md [hunk 2]
```diff
@@ -240,14 +242,14 @@ Additional **optional** attributes:
...
-|                  |                                                                               |          |                                                   |            | `hw.type` (required) | `memory`                                        |
+| Name             | Description                                                                   | Units    | Instrument Type ([*](README.md#instrument-types)) | Value Type | Attribute Key            | Attribute Values                                |
+| ---------------- | ----------------------------------------------------------------------------- | -------- | ------------------------------------------------- | ---------- | ------------------------ | ----------------------------------------------- |
+| `hw.errors`      | Number of errors encountered on this memory module                            | {errors} | Counter                                           | Int64      | `hw.type` (**Required**) | `memory`                                        |
```
Comment (jack-berg) [inline specification/metrics/semantic_conventions/hardware-metrics.md:247]: See #3298.
```suggestion
| `hw.errors`      | Number of errors encountered on this memory module                            | {error} | Counter                                           | Int64      | `hw.type` (**Required**) | `memory`                                        |
```

File: specification/metrics/semantic_conventions/hardware-metrics.md [hunk 3]
```diff
@@ -290,14 +292,14 @@ Additional **optional** attributes:
...
 | ---------------------------------------- | ------------------------------------------------------------------------------------------- | -------- | ------------------------------------------------- | ---------- | ------------------------------- | ----------------------------------------------- |
-| `hw.errors`                              | Number of errors encountered on this disk                                                   | {error} | Counter                                           | Int64      | `hw.type` (required)            | `physical_disk`                                 |
-| `hw.physical_disk.endurance_utilization` | Endurance remaining for this SSD disk                                                       | 1        | Gauge                                             | Double     | `state` (required)              | `remaining`                                     |
+| `hw.errors`                              | Number of errors encountered on this disk                                                   | {errors} | Counter                                           | Int64      | `hw.type` (**Required**)        | `physical_disk`                                 |
```
Comment (jack-berg) [inline specification/metrics/semantic_conventions/hardware-metrics.md:295]: See #3298.
```suggestion
| `hw.errors`                              | Number of errors encountered on this disk                                                   | {error} | Counter                                           | Int64      | `hw.type` (**Required**)        | `physical_disk`                                 |
```

File: specification/metrics/semantic_conventions/hardware-metrics.md [hunk 4]
```diff
@@ -332,14 +334,14 @@ Additional **optional** attributes:
...
+| Name                       | Description                                                                   | Units        | Instrument Type ([*](README.md#instrument-types)) | Value Type | Attribute Key            | Attribute Values                             |
+| -------------------------- | ----------------------------------------------------------------------------- | ------------ | ------------------------------------------------- | ---------- | ------------------------ | -------------------------------------------- |
+| `hw.errors`                | Number of errors encountered by the tape drive                                | {errors}     | Counter                                           | Int64      | `hw.type`                | `tape_drive`                                 |
+| `hw.tape_drive.operations` | Operations performed by the tape drive                                        | {operations} | Counter                                           | Int64      | `type` (Recommended)     | `mount`, `unmount`, `clean`                  |
```
Comment (jack-berg) [inline specification/metrics/semantic_conventions/hardware-metrics.md:340]: See #3298.
```suggestion
| `hw.tape_drive.operations` | Operations performed by the tape drive                                        | {operation} | Counter                                           | Int64      | `type` (Recommended)     | `mount`, `unmount`, `clean`                  |
```

**Review by jack-berg — COMMENTED** [review]
Just noticed that this reverts a recent change to units done in #3298.

**Comment by bertysentry** [comment]
@jack-berg Let's merge this, please! (I had to fix CHANGELOG.md once more)

BTW, I think we should drop CHANGELOG.md the way it's done currently. Maybe find a way to generate it automatically, based on merged PRs. I don't know, but it's a huge pain in the a\*\* for everybody: each time a change is merged (and therefore CHANGELOG.md is updated), you get a conflict in your hands.

**Review by jack-berg — APPROVED** [review]

**Pull request merged**